lib/know/src/classes/instant_message.rs

```diff
@@ -0,0 +1,129 @@
+// This is free and unencumbered software released into the public domain.
+
+use super::ThingLike;
+use crate::{
+    datatypes::{DateTime, InstantMessageHandle, InstantMessageId, PhoneNumber},
+    formatters::{
+        DisplayConcise, DisplayDetailed, DisplayInline, DisplayJsonLd, DisplayMime, DisplayOneliner,
+    },
+    prelude::Name,
+    traits,
+};
+use alloc::fmt;
+
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct InstantMessage {
+    pub id: Option<InstantMessageId>,
+    pub date: DateTime,
+    pub sender: InstantMessageHandle,
+    pub receiver: InstantMessageHandle,
+    pub platform: Option<String>,
+    pub references: Option<InstantMessageId>,
+    pub content: String,
+    // TODO: group context?
```

Comment:
The chat would contain the message, but sure, you could have an IFP (inverse functional property) backlink from the message to the chat as well